roles/etcd/templates/etcd.conf.j2

```diff
@@ -1,15 +1,15 @@
 {% macro initial_cluster() -%}
 {% for host in groups[etcd_peers_group] -%}
 {% if loop.last -%}
-{{ host }}={{ etcd_peer_url_scheme }}://{{ hostvars[host]['ansible_' + etcd_interface]['ipv4']['address'] }}:{{ etcd_peer_port }}
+{{ hostvars[host]['ansible_hostname'] }}={{ etcd_peer_url_scheme }}://{{ hostvars[host]['ansible_fqdn'] }}:{{ etcd_peer_port }}
```

Comment:
yes, they will propagate to hostvars (for that play and any plays that are run following it).

They will not automatically be propagated to facts, unless they are persisted somehow (using local_facts or some type of fact module that would need to be created/invoked).